### ISSUE TITLE
[AUDIT][SPARK-32614][SQL] Don't apply comment processing if 'comment' unset for CSV

### ISSUE THREAD

**Issue by abellina** [NVIDIA/spark-rapids#8205]
This was an audit for Spark 3.5 https://github.com/apache/spark/pull/29516 where the CSV parser was changed to not consider lines that start with a null character as comments. 

Likely this falls under the corner case category, but something we should at least investigate.

**Comment by revans2** [comment]
Yes this is probably a corner case, but we use the "null" character to turn off comments in CUDF.  So it might be more relevant than we think. At least we should test it and document any differences.
